rts/lra/coordinator/src/test/java/io/narayana/lra/coordinator/domain/model/LRATest.java

```diff
@@ -981,15 +979,40 @@ public void testGrandparentContext() {
         lraClient.closeLRA(grandParent); // should close everything in the hierarchy
 
         // nothing should be associated with the calling thread
-        assertNull("testGrandparentContext: current thread should not be associated with any LRAs",
-                lraClient.getCurrent());
-
         // and verify they are all closed
         assertStatus("grandparent", grandParent, null, LRAStatus.Closed);
         assertStatus("parent", parent, null, LRAStatus.Closed);
         assertStatus("child", child, null, LRAStatus.Closed);
     }
 
+    @Test
+    public void testParentLRAContext() {
+        // start a hierarchy of three LRAs
+        URI grandParent = lraClient.startLRA("NestedParticipantIT#testParentLRAContext grandparent");
+        URI parent = lraClient.startLRA("NestedParticipantIT#testParentLRAContext parent"); // child of grandParent
+        URI child = lraClient.startLRA("NestedParticipantIT#testParentLRAContext child"); // child of parent
+        lraClient.closeLRA(parent); // should close everything in the hierarchy
+        // verify the parent and child are closed
+        assertStatus("grandparent", grandParent, null, LRAStatus.Active);
+        assertStatus("parent", parent, null, LRAStatus.Closed);
+        assertStatus("child", child, null, LRAStatus.Closed);
+        lraClient.closeLRA(grandParent); // should close everything in the hierarchy
+    }
+
+    @Test
+    public void testNestedLRAContext() {
```

Comment:
The distinction in these tests is super subtle :)

Please can you help explain the difference a bit?

I can see the functional differences to be:
`NestedParticipantIT#testNestedLRAContext grandparent`
`"parent", parent, null, LRAStatus.Active`

Especially I am not clear why `lraClient.closeLRA(parent);` did not close the "parent"

Reply:
Thank you @tomjenkinson for pointing that out. That was a mistake, I've fixed it.
The idea is to test that when closing an LRA (also nested LRA) it needs to be is closed status as also all its children.

Reply:
There is a misunderstanding @tomjenkinson , when I created this PR the code was correct. I made the wrong change later ( with a force push, see https://github.com/jbosstm/narayana/compare/092374befa34cb8e516ccb0b087c2e6f54029d74..97ec0c0321bb78cac57d3d1e7db3750e819bf1eb). After that force push the build 575 started, but unfortunately another PR wan merged in the code base, so that build failed with "Narayana rebase on main failed. Please rebase it manually". So actually "closing the parent as I had it before" would have resulted in a failed build.

Reply:
Ah, thank you for clarifying!